libs/build_udunits.sh

```diff
@@ -37,19 +37,22 @@ export FCFLAGS="$FFLAGS"
 cd ${HPC_STACK_ROOT}/${PKGDIR:-"pkg"}
 
 software=$name-$version
-url=ftp://ftp.unidata.ucar.edu/pub/udunits/$software.tar.gz
-[[ -d $software ]] || ( $WGET $url; tar xvf $software.tar.gz && rm -f $software.tar.gz )
+gitTag="v${version}"
+gitURL=https://github.com/Unidata/UDUNITS-2.git
+
+[[ -d $software ]] || ( git clone -b $gitTag $gitURL $software)
 [[ ${DOWNLOAD_ONLY} =~ [yYtT] ]] && exit 0
 [[ -d $software ]] && cd $software || ( echo "$software does not exist, ABORT!"; exit 1 )
 [[ -d build ]] && rm -rf build
 mkdir -p build && cd build
 
-[[ $enable_shared =~ [yYtT] ]] && shared_flags="" || shared_flags="--disable-shared --enable-static"
+[[ $enable_shared =~ [yYtT] ]] && shared_flags=ON || shared_flags=OFF
 
-../configure --prefix=$prefix $shared_flags
+cmake .. \
+      -DCMAKE_INSTALL_PREFIX=$prefix \
+      -DBUILD_SHARED_LIBS=$shared_flags
 
 make -j${NTHREADS:-4}
-[[ "$MAKE_CHECK" = "YES" ]] && make check
```

Comment:
If there is a `ctest` option, we should run that in place of `make check`.  But, a check option should exist.

Reply:
Re-added, but `make test` is the correct option.

It won't run any tests though unless you have some C unit testing framework.

Reply:
how to do that?

Reply:
You need CUnit. When you run CMake it looks for it:

`-- Unable to find CUnit library. Unit-testing not enabled.`

Reply:
The reason I went with CMake was because the source code does not have `configure` file to run. You have to generate it yourself using `autoconf` and if you don't have all those libraries installed you won't get a `configure` script to run.

Reply:
sounds good. thanks for the info.